### ISSUE TITLE
[BigQuery] Subquery with where clause causing: Couldn't find closing bracket for opening bracket.

### ISSUE THREAD

**Issue by cjonesy** [sqlfluff/sqlfluff#869]
When running SQLFluff against a query containing subqueries I get `Couldn't find closing bracket for opening bracket.`. If I remove the `where` clause in the subquery then it will parse the query without erroring.

**Example query that does not parse:**
```sql
select
    table_a.id,
    b.some_name
from foo.table_a
left join (
    select
        id,
        some_name
    from (
        select id,
            some_name,
            row_number() over (partition by id order by inserted_at desc) as rn
        from foo.table_b
        where date(inserted_at) < current_date()
    )
    where rn = 1
    ) as b
    on table_a.id = b.id
```

**SQLFluff Output (with where clause):**
```bash
❯ sqlfluff lint --exclude-rules=L042 --dialect=bigquery --verbose ./foo.sql
==== sqlfluff ====
sqlfluff:           0.4.1 python:             3.9.0
dialect:         bigquery verbosity:              1

==== readout ====

=== [ path: ./foo.sql ] ===

== [foo.sql] FAIL
L:   5 | P:  11 |  PRS | Couldn't find closing bracket for opening bracket.
==== summary ====
violations:        1 status:         FAIL
```

**Example query that does parse:**
```sql
select
    table_a.id,
    b.some_name
from foo.table_a
left join (
    select
        id,
        some_name
    from (
        select id,
            some_name,
            row_number() over (partition by id order by inserted_at desc) as rn
        from foo.table_b
    )
    where rn = 1
    ) as b
    on table_a.id = b.id
```

**SQLFluff Output (without where clause):**
```bash
❯ sqlfluff lint --exclude-rules=L042 --dialect=bigquery --verbose ./foo.sql
==== sqlfluff ====
sqlfluff:           0.4.1 python:             3.9.0
dialect:         bigquery verbosity:              1

==== readout ====

=== [ path: ./foo.sql ] ===

==== summary ====
violations:        0 status:         PASS
```

**FWIW - SQLFluff v0.3.6 was able to parse the problematic query successfully:**
```bash
❯ sqlfluff lint --dialect=bigquery --verbose ./foo.sql
==== sqlfluff ====
sqlfluff:           0.3.6 python:             3.9.0
dialect:         bigquery verbosity:              1

==== readout ====

=== [ path: ./foo.sql ] ===


==== summary ====
violations:        0 status:         PASS
```

## Expected Behaviour
I expect SQLFluff to lint the query without erroring.

## Observed Behaviour
SQLFluff errors when it encounters the where clause within the subquery.

## Steps to Reproduce
Run SQLFluff against the example query shown above.

## Version
```bash
❯ sqlfluff --version
sqlfluff, version 0.4.1
❯ python --version
Python 3.9.0
```

## Configuration
Running with default config but overriding the following with args: `--exclude-rules=L042 --dialect=bigquery --verbose`

**Comment by cjonesy** [comment]
I just realized that this works in master, it was fixed after his commit: https://github.com/sqlfluff/sqlfluff/commit/993e7f4daf5a0b6b7c73b3405332f964ef121f25

Closing this out since it is no longer a bug!

**Comment by barrywhart** [comment]
🎉
